webapp/channels/src/components/list_modal.tsx

```diff
@@ -239,25 +246,22 @@ export default class ListModal extends React.PureComponent<Props, State> {
                             </div>
                             <div className='filter-controls'>
                                 {this.state.page > 0 &&
-                                <button
-                                    onClick={this.onPrev}
-                                    className='btn btn-sm btn-tertiary filter-control filter-control__prev'
-                                >
-                                    <FormattedMessage
-                                        id='filtered_user_list.prev'
-                                        defaultMessage='Previous'
+                                    <Button
+                                        onClick={this.onPrev}
+                                        emphasis='tertiary'
+                                        size='small'
+                                        label={messages.previous}
                                     />
-                                </button>}
-                                {this.props.numPerPage && (this.state.items.length >= this.props.numPerPage) && endCount !== this.state.totalCount &&
-                                <button
-                                    onClick={this.onNext}
-                                    className='btn btn-sm btn-tertiary filter-control filter-control__next'
-                                >
-                                    <FormattedMessage
-                                        id='filtered_user_list.next'
-                                        defaultMessage='Next'
+                                }
+                                {this.numPerPage && (this.state.items.length >= this.numPerPage) && endCount !== this.state.totalCount &&
```

Comment:
Why change this from `this.props.numPerPage` to `this.numPerPage`? I don't particularly understand why those both existed to begin with though

Reply:
I can move it to a different PR, but this was broken.

The reason for `this.numPerPage` to exist is to add a default to `this.props.numPerPage`, whenever 0 or undefined is passed. This is done on line 108. Also, it makes sure that changes in the props for `numPerPage` don't mess the pagination after the component is instantiated.

So... when used with a default `numPerPage`, the `next` button will never show.

We probably never hit this problem, because we don't have so many team or channel ldap groups (which is what this is used for if I am not mistaken).

That being said... this component is used in wrong ways at many levels. I invite you to see how it is used in `team_groups_manage_modal.tsx` or `channel_groups_manage_modal.tsx`. Among other things, what is happening is "a different component updating the state of this component directly".

The day we migrate this to function components, it is going to be fun (but really needed XD)